lib/systemd_testsuite_test.pm

```diff
@@ -70,15 +70,17 @@ sub testsuiteprepare {
     assert_script_run 'cd /var/opt/systemd-tests';
     assert_script_run "./run-tests.sh $testname --setup 2>&1 | tee /tmp/testsuite.log", 600;
     assert_script_run 'ls -l /etc/systemd/system/testsuite.service';
-    #reboot
-    power_action('reboot', textmode => 1);
-    wait_still_screen 90;
-    assert_screen('linux-login', 300);
-    type_string "root\n";
-    wait_still_screen 3;
-    type_password;
-    wait_still_screen 3;
-    send_key 'ret';
+    unless (check_var('BACKEND', 'qemu')) {
```

Comment:
AFAIK, this tests are expected to run for QEMU only. I don't know why that reboot was needed, but I would say this **unless** makes these code to be never executed.

Thomas.Blume@suse.com developed this tests, he should know. I will ask him.

Reply:
it can run on non-qemu, don;t know if anybody will run it there, but the code will not hurt there

Reply:
What I mean is that Thomas developed it to run it only on QEMU. Which means that this piece of code is expected to run on QEMU. And this change make it to not run in QEMU.

So, something is wrong there. The unless is wrong, or the code can be removed completely.

Reply:
Hm, actually, I wanted a reboot between each test in order to make sure that there are no remainders of previous tests affecting the current one.
When running the tests in qemu this is automatically done, since the test gets set up before qemu is started.
However, if you think this is too much reboot, let's just drop it.

Reply:
Thanks!
Then @dzedro was right.

Reply:
Just to double check, is the rollback to snapshot implemented for each architecture, e.g. ppc64le, aarch64 and s390x?

Reply:
rollback is supported on qemu backend (x86_64, ppc64le, aarch64), not on s390x

Reply:
@tblume, support for s390x depends on tumbleweed working on s390x (which is a project I help with openQA testing in my free time), so that we can build os-autoinst there with QEMU 3.* which has the virtual GPU for s390x that allows to use the QEMU backend of openQA.
So, it is in progress and expected.